lib/watch-files.js

```diff
@@ -1,19 +1,39 @@
 const chokidar = require('chokidar');
-const { isEmpty } = require('omnibelt');
+const { isEmpty, forEachSerialP, includes  } = require('omnibelt');
 const { logError, logResult, loadConfig, unlockConfig } = require('./utils');
 const paramsByCommand = require('./get-upload-params');
 const getUploader = require('./get-uploader');
 const path = require('path');
 
 module.exports = (nameOfDirectory) => {
   const uploader = getUploader(paramsByCommand[nameOfDirectory], { usePolling: process.platform === 'win32' });
+  const queue = [];
+  const queueData = new Map();
+  let isRunning = false;
+
   return async (command) => {
     const config = await loadConfig();
     if (isEmpty(config)) {
       return;
     }
+    const timerId = setInterval(async () => {
+      if (queue.length && !isRunning) {
+        logResult('upload', 'processing queue', 'magentaBright');
+        isRunning = true;
+        await forEachSerialP(async () => {
+          const file = queue.pop();
```

Comment:
use shift instead of pop to do FIFO. it feels weird to have things saved backwards for some reason. 
```
Started watching ::	experience
change	occurred on experience/components/gaTracking.hbs, queueing upload
change	occurred on experience/components/errorAlert.hbs, queueing upload
upload	processing queue
processing	experience/components/errorAlert.hbs
uploaded	experience/components/errorAlert.hbs
upload	processing queue
processing	experience/components/gaTracking.hbs
uploaded	experience/components/gaTracking.hbs
```